include/openssl/ssl.h

```diff
@@ -1395,6 +1395,31 @@ __owur int SSL_CIPHER_get_bits(const SSL_CIPHER *c, int *alg_bits);
 __owur char *SSL_CIPHER_get_version(const SSL_CIPHER *c);
 __owur const char *SSL_CIPHER_get_name(const SSL_CIPHER *c);
 __owur uint32_t SSL_CIPHER_get_id(const SSL_CIPHER *c);
+typedef enum {
+    SSL_CIPHER_KX_RSA = 1,
+    SSL_CIPHER_KX_ECDHE,
+    SSL_CIPHER_KX_DHE,
+    SSL_CIPHER_KX_ECDHE_PSK,
+    SSL_CIPHER_KX_DHE_PSK,
+    SSL_CIPHER_KX_RSA_PSK,
+    SSL_CIPHER_KX_PSK,
+    SSL_CIPHER_KX_SRP,
+    SSL_CIPHER_KX_GOST
```

Comment:
In fact, Steve (he's not on Github) just suggested as well that we should use/add NIDs. Lemme check and get back to you.